src/transactions/test/LiquidityPoolWithdrawTests.cpp

```diff
@@ -342,5 +343,93 @@ TEST_CASE("liquidity pool withdraw", "[tx][liquiditypool]")
             REQUIRE(acc1.getTrustlineBalance(poolNative1) == 0);
             checkLiquidityPool(*app, poolNative1, 0, 0, 0, 1);
         }
+
+        SECTION("large deposit/withdraw test")
+        {
+            struct Deposit
+            {
+                TestAccount acc;
+                int64_t numPoolShares;
+
+                Deposit(TestAccount const& acc, int64_t numPoolShares)
+                    : acc(acc), numPoolShares(numPoolShares)
+                {
+                }
+            };
+
+            std::vector<Deposit> deposits;
+
+            const int numAccounts = 100;
+
+            int64_t maxDeposit = INT64_MAX / numAccounts;
+            int64_t total = 0;
+
+            // deposit
+            for (int i = 0; i < numAccounts; ++i)
+            {
+                auto acc = root.create(fmt::format("account{}", i), minBal(10));
+                acc.changeTrust(cur1, INT64_MAX);
+                acc.changeTrust(cur2, INT64_MAX);
+                acc.changeTrust(share12, INT64_MAX);
+
+                std::uniform_int_distribution<int64_t> dist(1, maxDeposit);
+                int64_t size = dist(gRandomEngine);
+
+                root.pay(acc, cur1, size);
+                root.pay(acc, cur2, size);
+
+                acc.liquidityPoolDeposit(pool12, size, size, Price{1, 1},
+                                         Price{1, 1});
+
+                total += size;
+
+                checkLiquidityPool(*app, pool12, total, total, total, i + 1);
+
+                deposits.emplace_back(acc, size);
+            }
+
+            // trade through the pool
+            acc1.changeTrust(cur1, INT64_MAX);
+            acc1.changeTrust(cur2, INT64_MAX);
+            root.pay(acc1, cur1, 1000000);
+            root.pay(acc1, cur2, 1000000);
+
+            for (int i = 0; i < 100; ++i)
+            {
+                std::uniform_int_distribution<int64_t> dist(1, 1000);
+                int64_t sizeCur1 = dist(gRandomEngine);
```

Comment:
why not randomize (coin flip) to decide if it's cur1 or cur2 instead of always doing alternating trades?

Reply:
note that if you end up refactoring this into a subtest `InnerTest`, you probably want to not randomize inside the inner test (so that you fix the "swaps" and vary other parameters as to increase coverage)

Reply:
Done